Project.toml

```diff
@@ -17,6 +17,7 @@ KrylovKit = "0b1a1467-8014-51b9-945f-bf0ae24f4b77"
 LinearAlgebra = "37e2e46d-f89d-539d-b4ee-838fcccc9c8e"
 LinearMaps = "7a12625a-238d-50fd-b39a-03d52299707e"
 NDTensors = "23ae76d9-e61a-49c4-8f12-3f1a16adf9cf"
+Observers = "338f10d5-c7f1-4033-a7d1-f9dec39bcaa0"
```

Comment:
This reminds me that I would like to change `Observers` to be an optional dependency using a package extension, since `Observers` depends on `DataFrames` which is a relatively big package.

The idea would be that there would be a minimal interface for new observers to overload from `ITensorMPS` (say `ITensorMPS.update!`), and then users can load and pass an observer from `Observers.jl` into a solver. We wouldn't be able to use `Observers` directly inside of the package, but I think that is ok. Any use cases where I was tempted to do that I was able to use another code pattern, such as passing a `Ref` (or some other mutable data structure) as a keyword argument and writing a result into that `Ref`.

This would also have the advantage that we would have a better defined observer interface in `ITensorMPS` independent of `Observers.jl`, so in principle users could define and pass their own observer-like object. This could also help formalize how we can more conveniently keep backwards compatibility with the old observer system, without polluting `alternating_update`.

Not sure if we should do that in this PR or a follow-up, I think a follow-up would be fine, for now we are just forcing users to install an unnecessary dependency (unless they are using `ITensorTDVP.jl` with observers) but I'm sure most users won't notice.

Reply:
I added a note about this in the Google Doc.